contrib/install.sh

```diff
@@ -0,0 +1,1367 @@
+    --inputbox "Enter the BLOCK_HEIGHT you want to assume as valid (numeric value):" 10 60 \
+    3>&1 1>&2 2>&3)
+
+  check_dialog_escape
+
+  # Validate numeric input
+  if [[ "$block_height" =~ ^[0-9]+$ ]]; then
+    assume_valid="$block_height"
+    return 0
+  else
+    return 1
+  fi
+}
+
+# func: interactive_assume_utreexo
+#
+# Ask if user want to use --assume-utreexo or not
+interactive_assume_utreexo() {
+  if dialog --title "Floresta-Installer" --yesno "Do you want to use 'assume-utreexo'?\n\nThis option will significantly speed up the initial block download, by skipping the validation of the first hundreds of thousands of blocks. However, there's an inherent trust in the developer that the utreexo state is correct. Everything after the assumed height will be fully validated." 15 60; then
+    assume_utreexo=true
+  else
+    assume_utreexo=false
+  fi
+  check_dialog_escape
+  interactive_advanced_setup
+}
+
+# func: interactive_ssl
+#
+# Ask if user want to use SSL (Secure Sockets Layer) and if yes,
+# add --ssl-electrum-address, --ssl-cert-path and --ssl-key-path to florestad process.
+interactive_ssl() {
+  if dialog --title "Floresta-Installer" --yesno "Do you want to use SSL (Secure Sockets Layer) for the Electrum Server?\n\nThis is particularly important if you want to access the Electrum Server from a public untrusted network. This will add encryption and authentication to your service." 12 45; then
+    enable_ssl=true
+  else
+    enable_ssl=false
+  fi
+  check_dialog_escape
+  interactive_advanced_setup
+}
+
+# func: interactive_ask_for_add_xpubs
+#
+# Ask if user want to add one or more xpubs to our wallet. For each xpub, add it to the wallet_xpubs.
+# Keep asking until user do not want to add more xpubs or if an invalid one appears
+interactive_ask_for_add_xpubs() {
+  if dialog --title "Floresta-Installer (Ask XPUBs)" --yesno "Do you want to add one or more XPUBs to the node's wallet?" 8 45; then
+    while true; do
+      check_dialog_escape
+      interactive_add_xpub
+      code=$?
+      if [ "$code" -eq 1 ]; then
+        wallet_xpubs=()
+        dialog --title "Floresta-Installer" --msgbox "❌ Invalid XPUB detected. Restarting XPUB entry process." 8 45
+        return 1
+      elif [ "$code" -eq 2 ]; then
+        break
+      fi
+
+      if ! dialog --title "Add Another XPUB?" --yesno "Do you want to add another XPUB?" 8 45; then
+        break
+      fi
+    done
+  fi
+}
+
+# func: interactive_add_xpub
+#
+# Ask to user add a valid xpub. If a valid one is provided, add to wallet_xpubs array, show the
+# xpub to user confirm it and back to interactive_ask_for_add_xpubs function (thourgh returning true).
+# If user provided an invalid one, clean the wallet_xpubs list and back to interactive_ask_for_add_xpubs
+# to do all again.
+interactive_add_xpub() {
+  local xpub_regex="^(xpub|ypub|zpub|tpub)[A-Za-z0-9]{107}$" # (X/Y/Z/T)PUBS are 111 chars
+
+  while true; do
```

Comment:
If I select `Cancel`, it shows "Invalid Descriptor" and open the xpub input again